core/field_matrix.js

```diff
@@ -23,544 +23,550 @@
- * Called when the field is placed on a block.
- * @param {Block} block The owning block.
- */
-Blockly.FieldMatrix.prototype.init = function() {
-  if (this.fieldGroup_) {
-    // Matrix menu has already been initialized once.
-    return;
+    this.mouseDownWrapper_ = Blockly.browserEvents.bind(
+        this.getClickTarget_(), 'mousedown', this, this.onMouseDown_);
   }
 
-  // Build the DOM.
-  this.fieldGroup_ = Blockly.utils.createSvgElement('g', {}, null);
-  this.size_.width = Blockly.FieldMatrix.THUMBNAIL_SIZE +
-    Blockly.FieldMatrix.ARROW_SIZE + (Blockly.BlockSvg.DROPDOWN_ARROW_PADDING * 1.5);
-
-  this.sourceBlock_.getSvgRoot().appendChild(this.fieldGroup_);
-
-  var thumbX = Blockly.BlockSvg.DROPDOWN_ARROW_PADDING / 2;
-  var thumbY = (this.size_.height - Blockly.FieldMatrix.THUMBNAIL_SIZE) / 2;
-  var thumbnail = Blockly.utils.createSvgElement('g', {
-    'transform': 'translate(' + thumbX + ', ' + thumbY + ')',
-    'pointer-events': 'bounding-box', 'cursor': 'pointer'
-  }, this.fieldGroup_);
-  this.ledThumbNodes_ = [];
-  var nodeSize = Blockly.FieldMatrix.THUMBNAIL_NODE_SIZE;
-  var nodePad = Blockly.FieldMatrix.THUMBNAIL_NODE_PAD;
-  for (var i = 0; i < 5; i++) {
-    for (var n = 0; n < 5; n++) {
-      var attr = {
-        'x': ((nodeSize + nodePad) * n) + nodePad,
-        'y': ((nodeSize + nodePad) * i) + nodePad,
-        'width': nodeSize, 'height': nodeSize,
-        'rx': nodePad, 'ry': nodePad
-      };
-      this.ledThumbNodes_.push(
-          Blockly.utils.createSvgElement('rect', attr, thumbnail)
-      );
+  /**
+   * Set the value for this matrix menu.
+   * @param {string} matrix The new matrix value represented by a 25-bit integer.
+   * @override
+   */
+  setValue(matrix) {
+    if (!matrix || matrix === this.matrix_) {
+      return;  // No change
+    }
+    if (this.sourceBlock_ && Blockly.Events.isEnabled()) {
+      Blockly.Events.fire(new Blockly.Events.BlockChange(
+          this.sourceBlock_, 'field', this.name, this.matrix_, matrix));
     }
-    thumbnail.style.cursor = 'default';
+    matrix = matrix + FieldMatrix.ZEROS.substr(0, 25 - matrix.length);
+    this.matrix_ = matrix;
     this.updateMatrix_();
   }
 
-  if (!this.arrow_) {
-    var arrowX = Blockly.FieldMatrix.THUMBNAIL_SIZE +
-      Blockly.BlockSvg.DROPDOWN_ARROW_PADDING * 1.5;
-    var arrowY = (this.size_.height - Blockly.FieldMatrix.ARROW_SIZE) / 2;
-    this.arrow_ = Blockly.utils.createSvgElement('image', {
-      'height': Blockly.FieldMatrix.ARROW_SIZE + 'px',
-      'width': Blockly.FieldMatrix.ARROW_SIZE + 'px',
-      'transform': 'translate(' + arrowX + ', ' + arrowY + ')'
-    }, this.fieldGroup_);
-    this.arrow_.setAttributeNS('http://www.w3.org/1999/xlink',
-        'xlink:href', Blockly.mainWorkspace.options.pathToMedia +
-        'dropdown-arrow.svg');
-    this.arrow_.style.cursor = 'default';
+  /**
+   * Get the value from this matrix menu.
+   * @return {string} Current matrix value.
+   */
+  getValue() {
```

Comment:
Since we want to use `value_` instead this can be removed.

Reply:
Done